jwst/ami/tests/test_ami_interface.py

```diff
@@ -1,34 +1,79 @@
+import numpy as np
 import pytest
 
 import stpipe
+from stpipe import crds_client
 
 from stdatamodels.jwst import datamodels
 
 from jwst.ami import AmiAnalyzeStep
 
 
-def test_ami_analyze_cube_fail():
-    """Make sure ami_analyze fails if input is CubeModel (_calints)"""
-    model = datamodels.CubeModel((25, 19, 19))
+@pytest.fixture()
+def mock_nrm_reference_file(tmp_path, monkeypatch):
+    """
+    At the moment nrm only exists on CRDS-TEST so this
+    fixture will mock fetching any 'nrm' reference file
+    """
+    fn = tmp_path / "nrm.fits"
+
+    # make a fake nrm file
+    m = datamodels.NRMModel()
+    m.nrm = np.zeros((1024, 1024), dtype='f4')
+    m.save(fn)
+
+    original_get_reference_file = crds_client.get_reference_file
+
+    def mock_get_reference_file(dataset, reference_file_type, observatory):
+        if reference_file_type == 'nrm':
+            return str(fn)
+        return original_get_reference_file(dataset, reference_file_type, observatory)
+
+    monkeypatch.setattr(crds_client, "get_reference_file", mock_get_reference_file)
+
+
+@pytest.fixture()
+def example_model(mock_nrm_reference_file):
```

Comment:
Is it possible to also add a few random NaNs or large values here to add test coverage for all the bad pixel replacement, or is that too naive?

Reply:
That's a great question. Unfortunately my response is... maybe.

`example_model` here is rather unrepresentative of real data. I tried setting a `nan` value but there are a few issues.

One limitation at the moment is the lack of access to the `nrm` reference file on `crds` (it only exists on `crds-test` which the unit tests don't have access to). This would cause all the unit tests to fail, so I added a mock version ([above the example_model fixture](https://github.com/spacetelescope/jwst/pull/7862/files#diff-25ac33e148190671f89968855ac6fd4c6507a758584aaba2efba1c07e4691b35R13)) which at the moment creates a `NRMModel` with an empty (all zero) mask. This is also unrepresentative of the real reference file (but allows the step added in this PR to run).

I'm not sure if it's the faked mask or the faked data but if I add a `nan` value:
```
model.data[0, 30, 30] = np.nan
```
The `test_ami_analyze_step` fails with:
```
        # The fractional part of the offsets:
>       d0, d1 = (c[0] - int(c[0]), c[1] - int(c[1]))
E       ValueError: cannot convert float NaN to integer

jwst/ami/hextransformee.py:114: ValueError
```
If instead I use
```
model.data[:, 30, 30] = np.nan
```
The test passes and the nan-check in `bp_fix.find_bad_pixels` is covered but `bp_fix.fourier_corr` is not (due to the blank mask).

@rcooper295 is there a "minimal" model here of the mask and input data that you expect should pass with some bad pixel flagging?